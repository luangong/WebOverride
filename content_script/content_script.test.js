'use strict';

const { initialize, onMessage } = require('./content_script');

describe('Content script', () => {
  const location = {
    href: 'https://www.example.com/',
    reload: jest.fn()
  };

  describe('initialize()', () => {
    it('should not inject stuff into DOM', () => {
      initialize({}, document, location);
      expect(document.documentElement.outerHTML).toMatchSnapshot();
    });

    it('should not inject stuff into DOM', () => {
      initialize({
        '01234567-89ab-cdef-0123-456789abcdef': {
          active: true,
          urlQueries: [],
          htmlContent: '<div>foo</div>',
          cssContent: '.foo { color: #000; }',
          jsContent: 'console.log("bar")'
        }
      }, document, location);
      expect(document.documentElement.outerHTML).toMatchSnapshot();
    });

    it('should inject stuff into DOM', () => {
      initialize({
        '01234567-89ab-cdef-0123-456789abcdef': {
          active: true,
          urlQueries: ['^https://www\\.example\\.com/'],
          htmlContent: '<div>foo</div>',
          cssContent: '.foo { color: #000; }',
          libs: [{ value: 'https://code.jquery.com/jquery.min.js' }],
          jsContent: 'console.log("bar")'
        },
        '01234567-89ab-cdef-0123-456789abcdf0': {
          active: false,
          urlQueries: ['^https://www\\.example\\.com/'],
          htmlContent: '<div>foo</div>',
          cssContent: '.foo { color: #000; }',
          jsContent: 'console.log("bar")'
        },
        '01234567-89ab-cdef-0123-456789abcdf1': {
          active: true,
          urlQueries: ['^https://www\\.example\\.com/'],
        },
      }, document, location);
      expect(document.documentElement.outerHTML).toMatchSnapshot();
    });
  });

  describe('onMessage()', () => {
    it('should not call location.reload()', () => {
      onMessage({}, location);
      expect(location.reload).not.toHaveBeenCalled();
    });

    it('should not call location.reload()', () => {
      onMessage({ request: 'overrideUpdated' }, location);
      expect(location.reload).not.toHaveBeenCalled();
    });

    it('should not call location.reload()', () => {
      onMessage({ urlQueries: [] }, location);
      expect(location.reload).not.toHaveBeenCalled();
    });

    it('should not call location.reload()', () => {
      onMessage({ request: 'overrideUpdated', urlQueries: [] }, location);
      expect(location.reload).not.toHaveBeenCalled();
    });

    it('should not call location.reload()', () => {
      onMessage({
        request: 'overrideUpdated',
        urlQueries: ['^https://www\\.example\\.net/']
      }, location);
      expect(location.reload).not.toHaveBeenCalled();
    });

    it('should call location.reload()', () => {
      onMessage({
        request: 'overrideUpdated',
        urlQueries: [
          '^https://www\\.example\\.net/',
          '^https://www\\.example\\.com/'
        ]
      }, location);
      expect(location.reload).toHaveBeenCalled();
    });
  });
});
