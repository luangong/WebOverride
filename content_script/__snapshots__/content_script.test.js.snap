// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content script initialize() should inject stuff into DOM 1`] = `"<html><head><style type=\\"text/css\\">.foo { color: #000; }</style><script type=\\"text/javascript\\" src=\\"https://code.jquery.com/jquery.min.js\\"></script><script type=\\"text/javascript\\">console.log(\\"bar\\")</script></head><body><div>foo</div></body></html>"`;

exports[`Content script initialize() should not inject stuff into DOM 1`] = `"<html><head></head><body></body></html>"`;

exports[`Content script initialize() should not inject stuff into DOM 2`] = `"<html><head></head><body></body></html>"`;
